docs/appendices/release-notes/unreleased.rst

```diff
@@ -193,3 +193,24 @@ Fixes
 - Fixed a regression introduced in 5.3.0 which caused ``INSERT INTO`` statements
   with a ``ON CONFLICT`` clause on tables with generated primary key columns to
   fail with an ``ArrayIndexOutOfBoundsException``.
+
+- Added a workaround for an issue that caused an exception to be thrown when
+  inserting a non-array value into a column that is dynamically created by
+  inserting an empty array, ultimately modifying the type of the column and then
+  selecting this column by the row's id, for example::
+
+    CREATE TABLE t (id int primary key, o OBJECT(dynamic));
+    INSERT INTO t VALUES (1, {x=[]});
+    INSERT INTO t VALUES (2, {x={}});  /* this is the culprit statement, inserting an object onto an array typed column */
+
+    SELECT * FROM test WHERE id=1;
+    SQLParseException[Cannot cast object element `thisfieldshouldbeanarray` with value `[]` to type `object`]
+
+  after the fix::
+
+    SELECT * FROM t WHERE id=1;
+    +----+-------------+
+    | id | o           |
+    +----+-------------+
+    |  1 | {"x": null} |
```

Comment:
```suggestion
    |  1 | {"x": null} |
    |  2 | {"x": {}}   |
```

Reply:
that's not valid, we ask by id, `id = 1` so only one row is returned.